### ISSUE TITLE
Can't "Close Preferences without applying" deadlock

### ISSUE THREAD

**Issue by byjrack** [rancher-sandbox/rancher-desktop#5010]
### Actual Behavior

Selecting Discard just reopens the same window.

### Steps to Reproduce

Due to another issue with the [proxy implementation and GUI](https://github.com/rancher-sandbox/rancher-desktop/issues/5009), I was trying to change the proxy via rdctl with the Pref window open in the GUI.  That seemed to create a deadlock where the Pref window couldn't close, but rdctl couldn't change the settings either.  Ended up force quitting RD (which failed the rdctl change).

### Result

![image](https://github.com/rancher-sandbox/rancher-desktop/assets/26876044/c9edd41c-f15f-44a6-bdb3-bbd61162ef3a)

### Expected Behavior

Discarding changes just sends you back to the primary panel 

### Additional Information

Maybe a .lock file is needed as a mutex flag so if the GUI is editing prefs than rdctl can detect the lock and error rather than hang.  And if rdctl is editing prefs (should be far shorter than Pref panel) than lock and the GUI can check before opening the pref dialog.

### Rancher Desktop Version

1.9.0

### Rancher Desktop K8s Version

1.25.6

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

Windows

### Operating System / Build Version

19044.2965

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

system proxy offered by a PAC file and defaults to requiring authentication Negotiate,NTLM,Basic in that order.

**Comment by mook-as** [comment]
Hmm, odd, the _Discard_ option is supposed to just let you continuing to close the window (that is, it shouldn't be doing anything that would cause the pref window to change pages).  Do you happen to know what preference you changed, and at that point what unsaved changes were in the pref window? That might be able to help us reproduce the issue.

If you had debug logging turned on, `server.log` would have `propose` lines containing the settings as found in the dialog (uncommitted), and `background.log` would have `settings-update` lines containing the committed settings.

My attempt which didn't reproduce the bug:
- Open preferences
- Run `rdctl set --diagnostics.show-muted`
- Toggle the telemetry setting in the main prefs window
- Click on _Cancel_ in the pref dialog, causing the discard dialog to show up
- Click _Discard changes_.
- At that point, the pref window closed as expected.

**Comment by byjrack** [comment]
With the proxy bug on 1.9 things have generally been not acting as I would expect.  My attempt was to edit the proxy settings in the GUI via the pref pane, but because I was getting interrupted by the unhandled exception I decided to try rdctl.  When I set address rdctl just hung.  That's when i tried to close out the Pref panel and got in a loop in the GUI.

I do see the proposeSettings in server.log as well (along with unmasked creds <cough>).  My gut says that while I thought rdctl and the background process for pref pane were deadlocked I am guessing the looping was more tied to a proxy exception popup.  Basically both server and the rdctl post to server were blocked by the unhandled exception.  That unhandled exception is on the main thread it seems (and fires based on background proceses) so while it is up panes won't render and other nasty business.  

So I am thinking my experience was just timing.  I bet I was in the Cancel/Discard when the exception fired so when I selected the Discard it just couldn't switch back to the main panel and ended up in the loop.  So going to close this one as 99% sure it's related to #5009 and when that gets fixed these oddities will go away as well.
